### ISSUE TITLE
Multiple parallel lines not working when all connected to clock

### ISSUE THREAD

**Issue by jverkoey** [PaintYourDragon/p9813#2]
Hallo! I've successfully hooked the library up to a connected strand of 256 pixels. I am now trying to hook up two 256 pixel strands in parallel and am running into a problem.

The first 256 pixel strand is hooked up to tx, the second hooked up to RXD. Both strands are hooked up to the 5V PSU and the ftdi adapter is grounded to the PSU as well.

When I hook only the first strand to the clock, it receives data fine. When I hook only the second strand to the clock, it also receives data fine. When I hook both lines' clocks up though, neither strand receives data anymore. In the case of the demo app they no longer animate. I've also tried running the rgb app and sadly they don't change color when both connected to the clock. I'm at a complete loss as to why this might be the case :( Thoughts?

**Comment by jverkoey** [comment]
I grabbed some extra pixels and set up two parallel strands on a breadboard of 200 pixels each. I was able to get them both to animate in parallel in this configuration with the same wiring layout as the other pixels. I then swapped one strand out from the extras for one of the original strands and both strands still animated. I then swapped the last strand out and neither strand continued to update. Going to try isolating whether this happens with both strands or not.

**Comment by jverkoey** [comment]
Just confirmed that I can run in parallel with a combination of the breadboard and one of the original 256 pixels. For whatever reason when I try to run in parallel with both 256 pixel strands it refuses to work.

**Comment by jverkoey** [comment]
For additional information, I'm running all of my power over about 8 feet of 16-gauge wire to the two 256-pixel strand and splitting the power to every 50 pixels using t-connectors. I then have another 16-gauge wire running to the data line at the beginning of each 256-pixel strand.

I'm wondering if it's possible that I'm getting some sort of interference from the power/data lines running next to each other?

**Comment by PaintYourDragon** [comment]
Can you elaborate on "refuses to work"? Does one strand operate but the other remains dark, or are both strands dark?

What are your connections to the breakout board, and what library settings are you using?

**Comment by jverkoey** [comment]
When I send data over the ftdi breakout board, none of the pixels update
their state to reflect the new data. I'm using the standard breakout board
config.

On Dec 3, 2012, at 6:44 AM, Paint Your Dragon notifications@github.com
wrote:

Can you elaborate on "refuses to work"? Does one strand operate but the
other remains dark, or are both strands dark?

What are your connections to the breakout board, and what library settings
are you using?

—
Reply to this email directly or view it on
GitHubhttps://github.com/PaintYourDragon/p9813/issues/2#issuecomment-10955697.

**Comment by PaintYourDragon** [comment]
Okay. To confirm, using either one strand at a time, connected to TX (data) and CTS (clock), does work?

What's the distance between the FTDI breakout and the strand(s)?

**Comment by jverkoey** [comment]
Appx 8-10 feet of 16 awg speaker wire running the data and clock lines. I
have about 6-8 feet of the same kind of 16 awg wire running from the psu to
the lighting panel, where it splits into six shorter 18 awg power lines for
each of the 50 pixel strand t connectors.

On Dec 3, 2012, at 9:44 AM, Paint Your Dragon notifications@github.com
wrote:

Okay. To confirm, using either one strand at a time, connected to TX (data)
and CTS (clock), does work?

What's the distance between the FTDI breakout and the strand(s)?

—
Reply to this email directly or view it on
GitHubhttps://github.com/PaintYourDragon/p9813/issues/2#issuecomment-10963146.

**Comment by PaintYourDragon** [comment]
I have a hunch it might be the capacitance of the wires. That's a LOT of copper between the FTDI board and the first LED. Great for power, not so good for signals. Try using narrow gauge wire (24 to 30 ga) for the signals (keeping the heavy wire for power). Better still, try to reduce the distance between the FTDI board and the strands. If none of that cuts it, might need to use transistors to boost the signals.

**Comment by jverkoey** [comment]
And confirmed on the ftdi hookup config, though it doesn't matter whether
the strand is on the first or second ftdi data out. As soon as both of the
pixel strands are connected to the clock, the pixels stop updating. If I
remove either one of the strand clock cables from the clock connection, the
other strand begins updating again correctly.

On Dec 3, 2012, at 9:44 AM, Paint Your Dragon notifications@github.com
wrote:

Okay. To confirm, using either one strand at a time, connected to TX (data)
and CTS (clock), does work?

What's the distance between the FTDI breakout and the strand(s)?

—
Reply to this email directly or view it on
GitHubhttps://github.com/PaintYourDragon/p9813/issues/2#issuecomment-10963146.

**Comment by jverkoey** [comment]
Ahhh, ok I will try that :)

On Dec 3, 2012, at 10:08 AM, Paint Your Dragon notifications@github.com
wrote:

I have a hunch it might be the capacitance of the wires. That's a LOT of
copper between the FTDI board and the first LED. Great for power, not so
good for signals. Try using narrow gauge wire (24 to 30 ga) for the signals
(keeping the heavy wire for power). Better still, try to reduce the
distance between the FTDI board and the strands. If none of that cuts it,
might need to use transistors to boost the signals.

—
Reply to this email directly or view it on
GitHubhttps://github.com/PaintYourDragon/p9813/issues/2#issuecomment-10964095.

**Comment by jverkoey** [comment]
Out of genuine curiousity, why would a single panel of lights work but not both when they're both pretty far away with the 16 awg?

**Comment by jverkoey** [comment]
http://path.com/p/qrxj4

This is the setup I have. You can see the psu and the two panels. I'm going to go find some narrower gauge wire and then update this thread.

**Comment by PaintYourDragon** [comment]
The wires themselves act like capacitors...the more copper there is (whether length or girth), the more 'rounded' the signals are by the time they reach the other end. My best guess what's going on is that one wire has just low enough capacitance for a passable signal still to get through, but trying to drive two wires in parallel pushes it beyond some threshold.

**Comment by PaintYourDragon** [comment]
Is it at all humanly possible to mount the FTDI board on the frame and use a 12 foot USB cable?

**Comment by jverkoey** [comment]
Totally possible. I'm using this config for now because I want to easily
plug/unplug things while the panel is rotated against the wall (with the
lights facing forward). Distance will still be a challenge because there
are two more panels like this.

On Dec 3, 2012, at 10:36 AM, Paint Your Dragon notifications@github.com
wrote:

Is it at all humanly possible to mount the FTDI board on the frame and use
a 12 foot USB cable?

—
Reply to this email directly or view it on
GitHubhttps://github.com/PaintYourDragon/p9813/issues/2#issuecomment-10965269.

**Comment by PaintYourDragon** [comment]
Holy crap. I can't wait to see that lit! :smile:

**Comment by jverkoey** [comment]
Haha me too :p

**Comment by jverkoey** [comment]
OMG Hurray!! That was it! I replaced the 16 gauge wire with a 24 gauge for the data/clock lines and both strands are updating again!! omg omg omg pictures soon :D

**Comment by jverkoey** [comment]
First vid of the two tiles in action for the first panel (of three!): https://path.com/p/1hBeWp

Thanks so much for your help! It looks like I can't use an AWG 24 cable over about 4-5 feet long before it starts losing data, so I've ordered a 15ft USB and will be mounting the FTDI to the middle panel in the final configuration :)

**Comment by PaintYourDragon** [comment]
BRB, crapping pants.

**Comment by jverkoey** [comment]
I have a feeling that when I connect the other four panels that the clock line will be split between six lines and need each cable to be a sixth of the current length. I sadly need to make the cables at least 6 feet across each, so how would I go about amplifying the clock power so that it works over longer distances?

**Comment by PaintYourDragon** [comment]
I'd situate the FTDI device as close to the center as possible, and zig-zag the wiring for each panel so that the first LED is also nearest the center.

If that won't cut it, I'd try using a line buffer IC like a 74HC541 close to the FTDI chip. Connect the same clock output to six inputs on the chip, then run each of those to the LED strands.

**Comment by jverkoey** [comment]
I'll try to run the cables as close to the ftdi as I can then.

One more vid to end the night: http://twitter.yfrog.com/nfhxqdthrmsbinlxpyjuufmzz
Playin' with sine waves :D

**Comment by jverkoey** [comment]
I just tested hooking up a third line to the clock and have verified what I feared: for each cable I add the max length of all cables is roughly 1/n of a single cable's max length. This is getting me down to maybe a foot or so of cabling available for 12 feet of panelling.

You mentioned using something like the 74HC541, how does that work in this situation? Does it amplify the signal so that it can travel a farther distance?
